src/main/java/runhistoryplus/patches/PotionUseAndDiscardRunHistoryPatch.java

```diff
@@ -179,4 +183,33 @@ public int[] Locate(CtBehavior ctMethodToPatch) throws CannotCompileException, P
             }
         }
     }
+
+    @SpirePatch(clz = ObtainPotionAction.class, method = "update")
+    public static class ObtainPotionActionAddLogging {
+        public static class ObtainPotionActionAddLoggingExprEditor extends ExprEditor {
+            @Override
+            public void edit(MethodCall methodCall) throws CannotCompileException {
+                if (methodCall.getMethodName().equals("obtainPotion")) {
+                    methodCall.replace("{ $_ = $proceed($$); runhistoryplus.patches.PotionUseAndDiscardRunHistoryPatch.ObtainPotionActionAddLogging.playerObtainPotionWithLogging($_, this.potion); }");
+                }
+            }
+        }
+
+        @SpireInstrumentPatch
+        public static ExprEditor obtainPotionActionAddLoggingPatch() {
+            return new ObtainPotionActionAddLoggingExprEditor();
+        }
+
+        public static void playerObtainPotionWithLogging(boolean isPotionObtained, final AbstractPotion potion) {
+            if (isPotionObtained) {
+                CardCrawlGame.metricData.addPotionObtainData(potion);
+            }
+            else {
+                List<String> l = PotionDiscardLog.potion_discard_per_floor.get(PotionDiscardLog.potion_discard_per_floor.size() - 1);
+                if (l != null) {
+                    l.add(potion.ID);
```

Comment:
This is recording potions that were never obtained or even visible to the player, right? For example, if you play Alchemize with no free potions slots.

I think we should just ignore these and not log anything -- these potions just a weird detail of how the code was written (it would make more sense to first check for having open potion slots, and only then generate the potions, but I guess the devs didn't care).

Reply:
I agree that it's a little bit odd to log those and we can just omit them. I was just thinking of 'potions discarded' as 'potions not used effectively'.